exporter/elasticsearchexporter/config.go

```diff
@@ -81,7 +81,8 @@ type LogstashFormatSettings struct {
 }
 
 type DynamicIndexSetting struct {
-	Enabled bool `mapstructure:"enabled"`
+	Enabled bool   `mapstructure:"enabled"`
+	Mode    string `mapstructure:"mode"` // prefix_suffix or data_stream
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33794 introduced DS routing without dynamic index mode. Can you update this PR to take advantage of the change? It should be about changing the default dataset if otel mode is used.